de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/session/NegotiationHandler.java

```diff
@@ -25,65 +31,64 @@
   private static final Logger log = Logger.getLogger(NegotiationHandler.class);
 
   private final ISarosSessionManager sessionManager;
+  private final IWorkspace workspace;
   private final ThreadPoolExecutor sessionExecutor =
       new ThreadPoolExecutor(
           0,
           4,
           60,
           TimeUnit.SECONDS,
-          new LinkedBlockingQueue<Runnable>(),
+          new LinkedBlockingQueue<>(),
           new NamedThreadFactory("ServerSessionNegotiation-"));
   private final ThreadPoolExecutor projectExecutor =
       new ThreadPoolExecutor(
           0,
           10,
           60,
           TimeUnit.SECONDS,
-          new LinkedBlockingQueue<Runnable>(),
+          new LinkedBlockingQueue<>(),
           new NamedThreadFactory("ServerProjectNegotiation-"));
 
-  public NegotiationHandler(ISarosSessionManager sessionManager) {
+  public NegotiationHandler(ISarosSessionManager sessionManager, IWorkspace workspace) {
     sessionManager.setNegotiationHandler(this);
     this.sessionManager = sessionManager;
+    this.workspace = workspace;
   }
 
   @Override
   public void handleOutgoingSessionNegotiation(final OutgoingSessionNegotiation negotiation) {
 
     sessionExecutor.execute(
-        new Runnable() {
-          @Override
-          public void run() {
-            SessionNegotiation.Status status;
-            if (ServerConfig.isInteractive()) {
-              status = negotiation.start(new ConsoleProgressIndicator(System.out));
-            } else {
-              status = negotiation.start(new NullProgressMonitor());
-            }
-
-            switch (status) {
-              case OK:
-                sessionManager.startSharingProjects(negotiation.getPeer());
-                break;
-              case ERROR:
-                log.error("ERROR running session negotiation: " + negotiation.getErrorMessage());
-                break;
-              case REMOTE_ERROR:
-                log.error(
-                    "REMOTE_ERROR running session negotiation: "
-                        + negotiation.getErrorMessage()
-                        + " at remote: "
-                        + negotiation.getPeer().toString());
-                break;
-              case CANCEL:
-                log.info("Session negotiation was cancelled locally");
-                break;
-              case REMOTE_CANCEL:
-                log.info(
-                    "Session negotiation was cancelled by remote: "
-                        + negotiation.getPeer().toString());
-                break;
-            }
+        () -> {
```

Comment:
Is this just a refactor ? If this is just a refactor ... put this stuff into a method ! And not into a lambda